presto-main/src/main/java/com/facebook/presto/operator/MoreByteArrays.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.api.Experimental;
+
+import static com.facebook.presto.spi.JvmUtils.unsafe;
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+
+@Experimental
+public class MoreByteArrays
```

Comment:
@arhimondr hi Andrii, @mbasmanova and I specifically separated these two classes, because the methods in Unchecked class don't have range check. It is analogous to UncheckedBlock, where all methods don't have range check. Do we really want to merge them just to reduce the number of creation of unsafe?

Reply:
You can have them separate. It's just i really don't like the Unsafe being open.

Reply:
Ok. I created local private unsafe for UncheckedByteArrays and MoreByteArrays class